### PR TITLE
A different approach for testing of pexcept

### PR DESCRIPTION
**Pull request by Dipiert**
* Creating pexcept_config.h to be used in tests. This file will allow the definition of no-catch handlers and hooks.
* Bit more consistent hooks naming.
* Creating a new executable (test_pexcept_hooks) which will be have PEXCEPT_USE_CONFIG_FILE flag set to allow custom hooks and no-catch handlers in tests.
* Removing tests that need pexcept_config.h from test_pexcept.c (they live in test_pexcept_hooks now)
* Testing hooks and no-catch handler in test_pexcept_hooks.c

Still not getting the amount of commits correctly, sorry. After you merging this to prod I'll rebase my repo, push rebase and hopefully next PR will be OK.

### CODE REVIEW

File: test/CMakeLists.txt [hunk 1]
```diff
@@ -1,14 +1,12 @@
-add_executable(test_plist test_plist.c)
-add_executable(test_pstack test_pstack.c)
-add_executable(test_pqueue test_pqueue.c)
-add_executable(test_pdict test_pdict.c)
+include_directories(include)
+add_executable(test_pexcept_hooks ${CMAKE_SOURCE_DIR}/src/pexcept.c test_pexcept_hooks.c)
```
Comment (pbonsembiante) [inline test/CMakeLists.txt:2]: It'd be nice to have a comment explaining why this particular test suite needs special treatment.

File: test/CMakeLists.txt [hunk 2]
```diff
@@ -1,14 +1,12 @@
...
-target_link_libraries(test_pstack putils_static unity::framework)
-target_link_libraries(test_pqueue putils_static unity::framework)
-target_link_libraries(test_pdict putils_static unity::framework)
-
-add_test(TEST_PLIST ${EXECUTABLE_OUTPUT_PATH}/test_plist)
-add_test(TEST_PSTACK ${EXECUTABLE_OUTPUT_PATH}/test_pstack)
-add_test(TEST_PQUEUE ${EXECUTABLE_OUTPUT_PATH}/test_pqueue)
-add_test(TEST_PDICT ${EXECUTABLE_OUTPUT_PATH}/test_pdict)
+set(TEST_TARGETS test_plist test_pstack test_pqueue test_pdict test_pexcept)
+foreach(X IN LISTS TEST_TARGETS)
```
Comment (pbonsembiante) [inline test/CMakeLists.txt:8]: Consider renaming `X` to something more meaningful e.g `TARGET`

File: test/include/pexcept_config.h [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+#define PUTILS_PEXCEPT_CONFIG_H
+#endif
+
+#ifndef PEXCEPT_T
+#define PEXCEPT_T unsigned int
+#endif
+
+extern PEXCEPT_T uncaught_exception;
+extern short status;
+enum statuses {BEFORE_TRY, SUCCESS_TRY, AFTER_TRY, BEFORE_CATCH};
```
Comment (pbonsembiante) [inline test/include/pexcept_config.h:11]: Consider renaming this `enum` to its singular form, `status`, and possible add `_t` -> `status_t` . 

It usually confers a clear meaning when using the type in code, e.g: 

```c
int main(void) {
  enum statuses my_status = BEFORE_TRY;  //Note that the plural type name may hint that more than one status is possible at the same time
} 
```
vs

```c
int main(void) {
  enum status_t my_status = BEFORE_TRY; 
} 
```

File: test/test_pexcept.c [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ ***************************************************************************/
+
+#include "unity.h"
+#include "putils/pexcept.h"
+
+PEXCEPT_T uncaught_exception;
```
Comment (pbonsembiante) [inline test/test_pexcept.c:20]: This global variables appear unused, please remove them if that's the case

File: test/include/pexcept_config.h [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+
+#ifndef PEXCEPT_NO_CATCH_HANDLER
+#include <stdio.h>
```
Comment (pbonsembiante) [inline test/include/pexcept_config.h:14]: This seems unused ?

File: test/test_pexcept_hooks.c [hunk 6]
```diff
@@ -0,0 +1,66 @@
+#include "unity.h"
+#include "putils/pexcept.h"
+
+PEXCEPT_T uncaught_exception;
```
Comment (pbonsembiante) [inline test/test_pexcept_hooks.c:4]: Might be a good idea to set an initial default value to this two global variables 

```c
PEXCEPT_T uncaught_exception = PEXCEPT_NONE;
enum statuses status = NOTHING_HAPPENED;
```

**Review by pbonsembiante — CHANGES_REQUESTED** [review]

File: test/test_pexcept.c (hunk 4 above)
Comment (Dipiert) [reply test/test_pexcept.c]: solved :)

**Review by pbonsembiante — APPROVED** [review]
LGTM! +1

**Pull request merged**